### PR TITLE
Assignment8

### PR DESCRIPTION
**Pull request by ZStuefen**

### CODE REVIEW

**Comment by Thunderbird145** [comment]
I will review.

**Comment by Thunderbird145** [comment]
# Assignment
- Update the `GroupRepository` with functionality to generate gift assignments. ✔
  - A group with with 2 or fewer users should result in an error. This error should be displayed to a user. ✔
  - A user is not allowed to be both the Giver and Recipient of the assignment. ✔
- Update the GroupsController with an endpoint for generating the assignments. ✔
- Update the Groups/Edit page to contain the UI elements for the following
  - A means to generate assignments. ✔
  - A means to display any errors that occurs while generating assignments. ✔
  - Display of the assignments for each of the users. ✔
- Write appropriate unit tests for changes made to the `GroupRepository` ❌
- Write appropriate functional tests for the change made to the `GroupsController` ❌

## Extra Credit
- There is currently no way to for a user to add gifts. On the edit user page add functionality to be able to add a gift to the user. ❌
- Allow for navigating to a user from the Group edit page to make it easy to see the gifts requested by a user. ❌

**Comment by Thunderbird145** [comment]
looks like the unit tests are missing. Everything else appears to be present but I was unable to actually test it, running the website results in everything from the api server coming in as undefined. If it works on your end this might just be an issue I'm having, because everything looks okay as far as I can tell.

File: SecretSanta/src/SecretSanta.Api/Controllers/GroupsController.cs [hunk 1]
```diff
@@ -109,5 +109,19 @@ public ActionResult Add(int id, [FromBody] int userId)
             }
             return NotFound();
         }
+
+        [HttpPost("{id}/assign")]
+        [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+        [ProducesResponseType((int)HttpStatusCode.NotFound)]
```
Comment (Keboo) [inline SecretSanta/src/SecretSanta.Api/Controllers/GroupsController.cs:115]: The method does not appear to return a NotFound.
```suggestion
```

File: SecretSanta/src/SecretSanta.Business/GroupRepository.cs [hunk 2]
```diff
@@ -45,5 +45,25 @@ public void Save(Group item)
...
+        public AssignmentResult GenerateAssignments(int id)
+        {
+            bool res = MockData.Groups.TryGetValue(id, out Group? group);
+            if (res == false || group is null || group.Name is null || group.Name.Length == 0)
+            {
+                return AssignmentResult.Error("Group not found");
+            }
+            if (group.Users.Count <= 2)
+            {
+                return AssignmentResult.Error("Group " + group.Name + " must have at least three users");
```
Comment (Keboo) [inline SecretSanta/src/SecretSanta.Business/GroupRepository.cs:58]: C# has a cleaner way to build up strings called string interpolation.
```suggestion
                return AssignmentResult.Error($"Group {group.Name} must have at least three users");
```

**Review by Keboo — COMMENTED** [review]
# Assignment
- Update the `GroupRepository` with functionality to generate gift assignments. ✔
  - A group with with 2 or fewer users should result in an error. This error should be displayed to a user. ✔
  - A user is not allowed to be both the Giver and Recipient of the assignment. ✔
- Update the GroupsController with an endpoint for generating the assignments. ✔
- Update the Groups/Edit page to contain the UI elements for the following
  - A means to generate assignments. ✔
  - A means to display any errors that occurs while generating assignments. ✔
  - Display of the assignments for each of the users. ✔
- Write appropriate unit tests for changes made to the `GroupRepository` ❌
- Write appropriate functional tests for the change made to the `GroupsController` ❌

**Pull request closed without merging**